extensions/tika/runtime/src/main/java/io/quarkus/tika/TikaParser.java

```diff
@@ -0,0 +1,72 @@
+package io.quarkus.tika;
+
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.tika.io.TikaInputStream;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.parser.AutoDetectParser;
+import org.apache.tika.parser.ParseContext;
+import org.apache.tika.parser.Parser;
+import org.apache.tika.sax.ToTextContentHandler;
+import org.jboss.logging.Logger;
+import org.xml.sax.ContentHandler;
+import org.xml.sax.helpers.DefaultHandler;
+
+public class TikaParser {
+    private static final Logger LOG = Logger.getLogger(TikaParser.class.getName());
+
+    public Content getContent(InputStream stream) throws TikaParseException {
+        return getContent(stream, null);
+    }
+
+    public Content getContent(InputStream stream, String contentType) throws TikaParseException {
+        return createTikaContent(stream, contentType, new ToTextContentHandler());
+    }
+
+    public io.quarkus.tika.Metadata getMetadata(InputStream stream) throws TikaParseException {
+        return getMetadata(stream, null);
+    }
+
+    public io.quarkus.tika.Metadata getMetadata(InputStream stream, String contentType) throws TikaParseException {
+        return createTikaContent(stream, contentType,
+                contentType != null && contentType.contains("pdf") ? null : new DefaultHandler()).getMetadata();
+    }
+
+    protected Content createTikaContent(InputStream entityStream, String contentType, ContentHandler tikaHandler)
+            throws TikaParseException {
+        try {
+            Parser parser = createParser();
+            ParseContext context = new ParseContext();
+            context.set(Parser.class, parser);
+            org.apache.tika.metadata.Metadata tikaMetadata = new org.apache.tika.metadata.Metadata();
+            if (contentType != null) {
+                tikaMetadata.set(Metadata.CONTENT_TYPE, contentType);
+            }
+
+            try (InputStream tikaStream = TikaInputStream.get(entityStream)) {
+                parser.parse(tikaStream, tikaHandler, tikaMetadata, context);
+                return new Content(tikaHandler == null ? null : tikaHandler.toString().trim(), convert(tikaMetadata));
+            }
+        } catch (Exception e) {
+            LOG.warnf("%s stream can not be parsed", contentType);
```

Comment:
That's not how it should be done:
- you shouldn't have a warning
- you should have a proper message for your exception
- not sure if the `contentType` information is useful but in any case it might be an issue as if it's null, you will end up with `null stream can not be parsed` which is misleading. `Unable to parse stream for content-type ...` might be better with the last part conditioned to the `contentType` variable not being null.

Reply:
content-type will never be null at the MBR level, it is guaranteed by the JAX-RS spec, but your proposed message reads better, happy to update to it

Reply:
You're passing `null` as a `contentType` just a few lines above.

Reply:
Sorry, I forgot it was now not hidden any longer inside MBR :-)